woocommerce/woocommerce-android
# Fetch and display ad impressions forecast

Part of: #10574


### Description

Fetches and displays the ad impressions forecast

- Refactors how state is handle in `BlazeCampaignBudgetViewModel` from live data to a combination of StateFlows
- Fetches impressions forecast numbers after releasing the budget slider. 
- Shows loading state while fetching forecast results. 
- Handle errors and retry when failing to fetch impressions forecast. This matches iOS behavior. 

### Testing instructions

- Trigger Blaze campaign creation
- Open edit budget screen
- Play with budget slider and see how forecast is refreshed
- Remove connectivity and slide budget value, check the error is displayed
- Turn back on the internet connection and click `Retry`

### Images/gif



https://github.com/woocommerce/woocommerce-android/assets/2663464/5a51779c-e2c2-452a-9f4e-d4fbfc3d83a5
